app/uk/gov/hmrc/testppnsmultibox/services/TimeService.scala

```diff
@@ -18,19 +18,30 @@ package uk.gov.hmrc.testppnsmultibox.services
 
 import java.time.Clock
 import java.time.temporal.ChronoUnit.MINUTES
-import javax.inject.{Inject, Singleton}
+import javax.inject.{Inject, Named, Singleton}
 
+import akka.actor.ActorRef
+
+import uk.gov.hmrc.http.HeaderCarrier
+
+import uk.gov.hmrc.testppnsmultibox.actors.NotifyAtActor.NotifyAt
 import uk.gov.hmrc.testppnsmultibox.domain.models.TimedNotification
 import uk.gov.hmrc.testppnsmultibox.ppns.models.{BoxId, CorrelationId}
 import uk.gov.hmrc.testppnsmultibox.repositories.TimedNotificationRepository
 
 @Singleton()
-class TimeService @Inject() (uuidService: UuidService, timedNotificationRepository: TimedNotificationRepository, clock: Clock) {
+class TimeService @Inject() (
+    uuidService: UuidService,
+    timedNotificationRepository: TimedNotificationRepository,
+    @Named("notify-at-actor") notifyAtActor: ActorRef,
+    clock: Clock
+  ) {
 
-  def notifyMeIn(minutes: Int, boxId: BoxId): CorrelationId = {
+  def notifyMeIn(minutes: Int, boxId: BoxId)(implicit hc: HeaderCarrier): CorrelationId = {
     val correlationId = uuidService.correlationId
-    timedNotificationRepository.insert(TimedNotification(boxId, correlationId, notifyAt = clock.instant().plus(minutes, MINUTES)))
-    // TODO: Trigger asynchronous process
+    val notifyAt      = clock.instant().plus(minutes, MINUTES)
+    timedNotificationRepository.insert(TimedNotification(boxId, correlationId, notifyAt))
+    notifyAtActor ! NotifyAt(notifyAt, boxId, correlationId, hc)
```

Comment:
You could just use the scheduler in the actor system (in the play app) here instead of having to wire in the actor and such.

Reply:
I know, but I wanted to make the actor explicit here. This architecture means that it's ready for persisting the actor's queue to add more resilience (persisting the queue then reviving it after a restart) later.  I thought it would be easy to add the resilience, but it took longer than I expected to get to this point. I am planning to park that for another ticket. Incidentally, I was in two minds about where to put the Mongo persistence ... in the actor or before interacting with the actor. I think this way is safer.

Reply:
I've removed the actor as it is overkill if we're not doing resilience.